hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/record/AbstractNearCacheRecord.java

```diff
@@ -42,6 +42,7 @@
             AtomicLongFieldUpdater.newUpdater(AbstractNearCacheRecord.class, "recordState");
 
     protected long creationTime = TIME_NOT_SET;
+    protected int partitionId;
```

Comment:
I would like that, but I think this clashes with the actual reservation system. Maybe we can refactor this in another PR (Ahmet is also working on a fix on this, which already gets rid of some internal states etc.).

I'll make the fields volatile for now.